### PR TITLE
Fix IAM role deletion for AWS provisioner (resolves #1280)

### LINKED ISSUE #1280: AWS provisioner test leaks roles
http://imgur.com/a/8UlBi

### PR DESCRIPTION
**Pull request by cket**
First commit just modifies the test to expose this bug

### CODE REVIEW

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
@hannes-ucsc ready for review

**Comment by cket** [comment]
@hannes-ucsc there is one profile and role per cluster

**Comment by cket** [comment]
@hannes-ucsc ready for review

**Comment by cket** [comment]
jenkins, test this please

File: src/toil/test/provisioners/aws/awsProvisionerTest.py [hunk 1]
```diff
@@ -43,12 +46,20 @@ def tearDown(self):
         from toil.provisioners.aws.awsProvisioner import AWSProvisioner
         AWSProvisioner.destroyCluster(self.clusterName)
 
+    def getMatchingRoles(self, roleName):
+        ctx = Context(self.awsRegion()+'a', '/')
+        roles = ctx._get_all_roles()
+        roleName = roleName.replace('-', '_')
```
Comment (hannes-ucsc) [inline src/toil/test/provisioners/aws/awsProvisionerTest.py]: Is this working around a quirk in CGCloud?

File: src/toil/test/provisioners/aws/awsProvisionerTest.py [hunk 2]
```diff
@@ -43,12 +46,20 @@ def tearDown(self):
         from toil.provisioners.aws.awsProvisioner import AWSProvisioner
         AWSProvisioner.destroyCluster(self.clusterName)
 
+    def getMatchingRoles(self, roleName):
+        ctx = Context(self.awsRegion()+'a', '/')
```
Comment (hannes-ucsc) [inline src/toil/test/provisioners/aws/awsProvisionerTest.py]: Do we really want to hard-code the zone here?

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Pull request merged**